use-cases/school-timetabling/src/main/java/org/acme/schooltimetabling/rest/TimetableResource.java

```diff
@@ -102,23 +103,45 @@ public String solve(Timetable problem) {
     @Produces(MediaType.APPLICATION_JSON)
     @Path("{jobId}")
     public Timetable getTimeTable(
-            @Parameter(description = "The job ID returned by the POST method.") @PathParam("jobId") String jobId,
-            @QueryParam("retrieve") Retrieve retrieve) {
-        retrieve = retrieve == null ? Retrieve.FULL : retrieve;
+            @Parameter(description = "The job ID returned by the POST method.") @PathParam("jobId") String jobId) {
+        Timetable timetable = getTimetableAndCheckForExceptions(jobId);
+        SolverStatus solverStatus = solverManager.getSolverStatus(jobId);
+        timetable.setSolverStatus(solverStatus);
+        return timetable;
+    }
+
+    @Operation(
+            summary = "Get the timetable status and score for a given job ID.")
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", description = "The timetable status and the best score so far.",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON,
+                            schema = @Schema(implementation = Timetable.class))),
+            @APIResponse(responseCode = "404", description = "No timetable found.",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON,
+                            schema = @Schema(implementation = ErrorInfo.class))),
+            @APIResponse(responseCode = "500", description = "Exception during solving a timetable.",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON,
+                            schema = @Schema(implementation = ErrorInfo.class)))
+    })
```

Comment:
This is a massive obfuscation of *quickstart* code. I wouldn't add these.

Reply:
I see your point, but in my opinion it goes against the goal of reaching similarity between QS and models. Let's discuss this later with Geoffrey.